docs-client/src/containers/App/index.tsx

```diff
@@ -88,127 +91,155 @@ const styles = (theme: Theme) =>
 interface State {
   mobileDrawerOpen: boolean;
   specification?: Specification;
+  servicesOpen: boolean;
+  enumsOpen: boolean;
+  structsOpen: boolean;
+  exceptionsOpen: boolean;
 }
 
 type Props = WithStyles<typeof styles> & RouteComponentProps<{}>;
 
 interface AppDrawerProps extends WithStyles<typeof styles> {
   specification: Specification;
   navigateTo: (url: string) => void;
+  state: State;
+  handleCollapse: (itemName: string) => void;
 }
 
-function AppDrawer({ classes, navigateTo, specification }: AppDrawerProps) {
+function AppDrawer({
+  classes,
+  navigateTo,
+  specification,
+  state,
+  handleCollapse,
+}: AppDrawerProps) {
   return (
     <List component="nav">
       {specification.getServices().length > 0 && (
         <>
-          <ListItem>
+          <ListItem button onClick={() => handleCollapse('services')}>
             <ListItemText disableTypography>
               <Typography variant="headline">Services</Typography>
             </ListItemText>
+            {state.servicesOpen ? <ExpandLess /> : <ExpandMore />}
           </ListItem>
-          {specification.getServices().map((service) => (
-            <div key={service.name}>
-              <ListSubheader className={classes.methodHeader}>
-                <Typography variant="subheading">
-                  <code>{simpleName(service.name)}</code>
-                </Typography>
-              </ListSubheader>
-              {service.methods.map((method) => (
-                <ListItem
-                  key={`${service.name}/${method.name}`}
-                  button
-                  onClick={() =>
-                    navigateTo(`/methods/${service.name}/${method.name}`)
-                  }
-                >
-                  <ListItemText
-                    inset
-                    primaryTypographyProps={{
-                      variant: 'body1',
-                    }}
+          <Collapse in={state.servicesOpen} timeout="auto" unmountOnExit>
```

Comment:
Let's remove `unmountOnExit` throughout. I guess while it is pretty complicated DOM, it's not unreasonable so the optimization of unmounting may not be worth the fact that the sidebar width changes when collapsing if we do unmount.

Reply:
You are right! It's better to keep the contents in DOM but just collapse.
Using `unmountOnExit `  may cause the sidebar width change which looks a little bit wired